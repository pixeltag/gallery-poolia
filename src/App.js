import './assets/css/App.css';
import Header from './layout/Header'
import Gallery from "./pages/Gallery";

function App() {
  return (
    <div className="App">
      <Header />
      <Gallery />
    </div>
  );
}

export default App;
