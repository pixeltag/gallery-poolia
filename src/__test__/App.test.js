import {
  render,
  screen,
  cleanup,
} from "@testing-library/react";
import { unmountComponentAtNode } from "react-dom";

import App from "./../App";

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  cleanup();
  container.remove();
  container = null;
});

test("App component renders correctly", async () => {
  const container = render(<App />);
  const mainElement = await container.findByRole("main");
  expect(mainElement).toBeInTheDocument();
});
