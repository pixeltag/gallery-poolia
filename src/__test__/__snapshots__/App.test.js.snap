// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery component should match the snapshot 1`] = `"<div class=\\"App\\"><header class=\\"header\\"><h1 class=\\"logo-text\\">Poolia Gallery</h1></header><main><section class=\\"gallery-container\\"><h2 class=\\"hide\\">Gallery Section</h2><div class=\\"not-found\\">Loading</div></section><div data-testid=\\"modal\\" class=\\"modal\\"><button class=\\"close-btn\\"><img src=\\"close.svg\\" alt=\\"Close modal\\"/></button><img data-testid=\\"img\\" class=\\"modal-img\\"/><div class=\\"modal-body\\"><h3 data-testid=\\"title\\">Image title</h3><p data-testid=\\"description\\"></p></div></div></main></div>"`;
